src/main/java/frc/robot/utilities/LEDSubStrip.java

```diff
@@ -0,0 +1,164 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+// THis class is a wrapper on AddressableLEDBuffer that allows logical divisions of one continuous
+// strip of LEDs
+
+package frc.robot.utilities;
+
+import edu.wpi.first.wpilibj.AddressableLEDBuffer;
+import edu.wpi.first.wpilibj.util.Color;
+import edu.wpi.first.wpilibj.util.Color8Bit;
+
+public class LEDSubStrip {
+  private int begin;
+  private int end;
+  private boolean direction;
+  private AddressableLEDBuffer stripBuffer;
+
+  // Full constructor with data buffer end, begining, and direction
+  public LEDSubStrip(AddressableLEDBuffer stripBuffer, int begin, int end, boolean direction) {
+    this.begin = begin;
+    this.end = end;
+    this.direction = direction;
+    this.stripBuffer = stripBuffer;
+  }
+
+  // Overloaded constructor with default direction
+  public LEDSubStrip(AddressableLEDBuffer stripBuffer, int begin, int end) {
+    this.begin = begin;
+    this.end = end;
+    this.direction = true;
+    this.stripBuffer = stripBuffer;
+  }
+
+  // Returns strip data to update physical strip
+  public AddressableLEDBuffer getData() {
+    return stripBuffer;
+  }
+
+  /**
+   * Sets a specific led in the section.
+   *
+   * @param index the index to write
+   * @param r the r value [0-255]
+   * @param g the g value [0-255]
+   * @param b the b value [0-255]
+   */
+  public void setRGB(int index, int r, int g, int b) {
```

Comment:
You could consider adding a simple helper method to always do the proper indexing, so that the offset calculation only occurs in one place. E.g.

```
private int getOffset(int index) {
    if (direction) return index + begin;
    return end - index;
}
```
And then the body of this method and also the two `getX()` methods below becomes simpler.
```
public void SetRGB(int index, int r, int g, int b) {
    stripBuffer.SetRGB(this.getOffset(index), r, g, b);
}
```

Reply:
The duplication of this (admittedly simple) calculation throughout the class seems like a code smell to me.

Reply:
Yeah tbh I wasn't bothered because not even all of the methods need it because a few of them internally call setRGB so it only has to be done there and in two others, but with the added benefit of dropping the if/else for direction I think'll implement this.